src/main/java/com/hedvig/claims/query/ClaimsEventListener.java

```diff
@@ -167,6 +173,36 @@ public void on(ClaimsTypeUpdateEvent e) {
         claim.addEvent(ev);
 
         claimRepository.save(claim);
+
+        setDefaultClaimDate(claim.id);
+    }
+
+    private void setDefaultClaimDate(String claimId) {
+        ClaimEntity claimEntity = claimRepository.findById(claimId).orElse(null);
```

Comment:
You'll already have the entity at this point
```suggestion
        setDefaultClaimDate(claim);
    }

    private void setDefaultClaimDate(ClaimEntity claim) {
```